### PR TITLE
Add support for AccuracyAuthorization

### PR DESCRIPTION
**Pull request by mackoj**
Fix for https://github.com/pointfreeco/swift-composable-architecture/issues/306

https://developer.apple.com/documentation/corelocation/claccuracyauthorization

- [x] Fix compiling issue with on macOS

### CODE REVIEW

**Comment by mackoj** [comment]
If someone has an idea how to fix the macOS build I'm listening...

**Comment by andreyz** [comment]
@mackoj I think you need to add 2 lines with compiler conditionals to hide CLAccuracyAuthorization from swift 5.2 compiler:, like so

``` swift
public enum AccuracyAuthorization {
  case fullAccuracy
  case reducedAccuracy

  #if compiler(>=5.3)
  @available(iOS 14, macCatalyst 14, macOS 11, tvOS 14, watchOS 7, *)
  init?(_ accuracyAuth: CLAccuracyAuthorization?) {
    switch accuracyAuth {
    case .fullAccuracy:
      self = .fullAccuracy
    case .reducedAccuracy:
      self = .reducedAccuracy
    default:
      return nil
    }
  }
  #endif
}
```

**Comment by mackoj** [comment]
Thanks I will it

**Comment by mackoj** [comment]
Hi @andreyz,

It don't seems to work either with this modification.

**Comment by mackoj** [comment]
```swift
#if compiler(>=5.3)
```

or

```swift
#if swift(>=5.3)
```

Don't seems to work either.

**Comment by stephencelis** [comment]
Thanks for the PR, @mackoj! We haven't had time to dig into the failure yet but we'll try to get to it soon.

**Comment by andreyz** [comment]
> ```swift
> ```swift
> #if compiler(>=5.3)
> ```
> 
> 
> or
> ```swift
> #if swift(>=5.3)
> ```
> 
> 
> Don't seems to work either.
> ```

We were almost there, we needed `#if compiler(>=5.3.1)` based on https://en.wikipedia.org/wiki/Xcode#Toolchain_versions

**Comment by mackoj** [comment]
@andreyz Thanks for the fix

File: Sources/ComposableCoreLocation/Interface.swift [hunk 1]
```diff
@@ -395,6 +399,10 @@ public struct LocationManager {
 
   public func location(id: AnyHashable) -> Location? { self.location(id) }
 
+  public func accuracyAuthorization(id: AnyHashable) -> AccuracyAuthorization? {
```
Comment (stephencelis) [inline Sources/ComposableCoreLocation/Interface.swift:402]: One small change. Let's guard the client from calling code that is unavailable:
```suggestion
  @available(iOS 14.0, tvOS 14.0, watchOS 7.0, macOS 11.0, macCatalyst 14.0, *)
  public func accuracyAuthorization(id: AnyHashable) -> AccuracyAuthorization? {
```

**Comment by stephencelis** [comment]
Looks great, thanks again!

**Pull request merged**